app/models/topic_list.rb

```diff
@@ -78,7 +78,7 @@ def topics
 
   def categories
     @categories ||=
-      topics.map { |t| [t.category, t.category&.parent_category] }.uniq.flatten.compact
+      topics.map { |t| [t.category&.parent_category, t.category] }.uniq.flatten.compact
```

Comment:
Curious why the order has to be changed here.

Reply:
The order categories are sorted matters for the frotend to ensure that the `parentCategory` properties are set correctly.

Reply:
I fixed that in another PR: https://github.com/discourse/discourse/pull/24530